8.x/theme-development/client-side-cart-module.md

```diff
@@ -0,0 +1,665 @@
+## Client Side cart module
+
+[[toc]]
+
+Larammerce base theme has a build-in JS module named **LocalCartService** to manage the client-side cart in the customer's browser.
+This document reviews the life cycle and progress of this module.
+
+
+**LocalCartService** is placed in the `resources/assets/js/define/local_cart_service.js` file consisting of 437 lines of code. It is based on require js module. To read more about it, refer to this address:[requirejs](https://www.requirejs.org/).
+
+First, let's check how **LocalCartService** manages the shopping cart:
+
+The image below is one of the websites created with Larammerce, and it can be seen that if the user is not logged in, the module manages the shopping cart, and if the user selects a product, it is added to the shopping cart.
+
+As you should know, when the user is not logged in, the shopping cart data is stored as JSON in the cookie and the data is stored on the server if the user is logged in.
+
+![LocalCartService.png](/LocalCartService.png)
+
+To see how the products are stored in the shopping cart, copy the value of local_cart_v11, and after decoding, the following value will be displayed:
```

Comment:
The coding algorithm should be mentioned.
Something like this:
To see how the products are stored in the shopping cart on the client-side, copy the value of the `local_cart_vn` (as `n` is the version of the client cookie management), decode it as it's encoded by default by the browser in url_encode format. As a result the following value should be resulted (A JSON object with product ids as the key and value filled by the count of each product.):